apps/web/src/data/subgraph/requests/tokenQuery.ts

```diff
@@ -0,0 +1,65 @@
+import { ethers } from 'ethers'
+import isUndefined from 'lodash/isUndefined'
+import omitBy from 'lodash/omitBy'
+
+import { sdk } from 'src/data/subgraph/client'
+
+export interface Token {
+  id: string
+  name?: string
+  image?: string
+  description?: string
+  owner?: string
+  mintDate?: string
+}
+
+export interface TokenWinner {
+  highestBidder?: string
+  price?: number
+}
+
+export const tokenQuery = async (
+  address: string,
+  tokenId: string
+): Promise<Token | undefined> => {
+  if (!address) return
+
+  const data = await sdk.token({ id: `${address.toLowerCase()}:${tokenId}` })
+
+  const token = data?.token
+
+  if (!token) {
+    return undefined
+  }
+
+  return {
+    id: token.tokenId,
+    ...omitBy(
+      {
+        owner: token.owner || undefined,
+        name: token.name || undefined,
+        description: token.dao.description || undefined,
```

Comment:
do we need the undefined fallback assignment if these values are already possibly undefined?